### DISCUSSION TITLE
Is there a way to get all mapped attributes for all groups/roles in one API request?

### DISCUSSION THREAD

**Discussion by NikKozh** [keycloak/keycloak#24377] [Q&A]
Hi, so in Keycloak we have an API request `GET /{realm}/users`, where for all users we are getting `attributes` field, which have all LDAP attributes according to `user-attribute-ldap-mapper` mappers.

For groups we also have `group-ldap-mapper` mapper and API request `GET /{realm}/groups`, which gives only name mapped to LDAP attribute as it set in "Group Name LDAP Attribute" setting. But to get all attributes for all groups, we have to call `GET /{realm}/groups/{id}` for every single existing group, which can be many, many requests.

So, is there a way to optimize this and get all attributes for all groups in single API request, like for users?

**Comment by PBAsydney** [comment]
Include ` {"briefRepresentation": False} ` in the query parameters, then the result will include attributes.

**Comment by mjsir911** [comment]
@PBAsydney `briefRepresentation` isn't working for me and has a lot of confusion surrounding it it seems, and not much documentation to boot. 

I have a user here, returned by the `/users` api endpoint:

```
  {
    "access": {
      "manage": false
    },
    "attributes": {
      "LDAP_ENTRY_DN": [
        "uid=m,cn=users,cn=accounts,dc=devhack,dc=net"
      ],
      "LDAP_ID": [
        "e570bd74-5bf8-11ef-8b78-bc2411a1700c"
      ],
      "createTimestamp": [
        "20240816175643Z"
      ],
      "mailcow_email": [
        "m@devhack.net"
      ],
      "modifyTimestamp": [
        "20251128021924Z"
      ],
      "secondary-email": [
        "m@sirabella.org"
      ]
    },
    "createdTimestamp": 1723832148267,
    "disableableCredentialTypes": [],
    "email": "m@devhack.net",
    "emailVerified": true,
    "enabled": true,
```

and even though this user is in a group with the `mailcow_template` attribute set on the group that contains the user, it's not being set on the user itself even when I pass `&briefRepresentation=False` in the querystring
